### PR TITLE
add support for multivalue/singlevalue tags

### PR DESCRIPTION
**Pull request by cianBuckley**
### Relevant information

Adds support for single value tags

### Checklist

* [ ] I've read the guidelines and understand the acceptance criteria.
* [ ] The value of the attribute marked as `identifier` will be unique and valid. 
* [ ] I've confirmed that my entity type wasn't already defined. If it is I'm providing an
 explanation above.

### CODE REVIEW

File: definitions/infra-container/definition.yml [hunk 1]
```diff
@@ -12,12 +12,14 @@ synthesis:
...
-    - container.state
-    - docker.state
-    - docker.containerId
-    - newrelic.integrationName
-    - newrelic.integrationVersion
-    - newrelic.agentVersion
+    - attribute: container.state
+      mutliValue: false
+    - attribute: docker.state
+      mutliValue: false
```
Comment (cianBuckley) [inline definitions/infra-container/definition.yml:18]: I think these are solid use cases for it, but just want to highlight these lines

File: README.md [hunk 2]
```diff
@@ -95,6 +95,13 @@ During synthesis, the tags will be created using the attribute name as key and i
 
 If some of the tags' attributes are not present on the telemetry message received, the entity will still be synthesized with the available tags (if any).
 
+```yaml
+  tags:
+    - attribute: attributeNameB
+      multiValue: false # declare if the value for this tag should be a list of string values (accumulated) or a single value (replaced). defaults to true
```
Comment (srvaroa) [inline README.md:101]: @jfjoly from your perspective does this sound meaningful, or would append / replace work better?

File: definitions/infra-container/definition.yml (hunk 1 above)
Comment (srvaroa) [reply definitions/infra-container/definition.yml:18]: s/mutliValue/multiValue

File: README.md [hunk 3]
```diff
@@ -95,6 +95,13 @@ During synthesis, the tags will be created using the attribute name as key and i
 
 If some of the tags' attributes are not present on the telemetry message received, the entity will still be synthesized with the available tags (if any).
 
+```yaml
+  tags:
+    - attribute: attributeNameB
```
Comment (srvaroa) [inline README.md:100]: Is is worth adding the attribute keyword? if we just leave `attributeNameB` it would also work, but being more concise?

Comment (cianBuckley) [reply README.md:100]: its an object so i think it needs it

Comment (srvaroa) [reply README.md:100]: This is how we do it in golden metrics:

https://github.com/newrelic-experimental/entity-synthesis-definitions/blob/main/definitions/infra-awsapigatewayapi/golden_metrics.yml#L9

Each of these are objects, but the name of the metric functions as identifier:

image

Comment (cianBuckley) [reply README.md:100]: Personally, I prefer it the way it is. From a clients perspective I don't think either way has much impact.
from an internal perspective, IMO it it is easier to operate on in code like such:
```
class Tag(attribute, multiValue = true)
List<Tag>
```
as apposed to 
```
class TagProperties(multiValue=true)
List<Map<String, TagProperties>
```

**Review by IreneMLC — APPROVED** [review]
LGTM

File: README.md (hunk 3 above)
Comment (srvaroa) [reply README.md:100]: @jfjoly any preference?

**Comment by naxhh** [comment]
Im closing this one and I'll open a new PR. We are going with a different format and there are other PR's that will force us to resolve conflicts so it's easier for me to open a new one :D

**Pull request closed without merging**